ta/arch/arm/link.mk

```diff
@@ -25,7 +25,7 @@ reverse = $(if $(wordlist 2,2,$(1)),$(call reverse,$(wordlist 2,$(words $(1)),$(
 link-ldadd  = $(LDADD)
 link-ldadd += $(addprefix -L,$(libdirs))
 link-ldadd += $(addprefix -l,$(call reverse,$(libnames)))
-ldargs-$(binary).elf := $(link-ldflags) $(objs) $(link-ldadd) $(libgcc$(sm))
```

Comment:
We're not building OK on STM platform, since we use the new optee_test on GitHub that make use of the devkit.

You can reproduce the build issue on vexpress by replacing, in `core/arch/arm/plat-vexpress/platform_flags.mk`, cpu flag `cortex-a15` into `cortex-a9` (the one of plat-stm). Compilation error in the tests is

     LD      /home/brand/devel/optee/out/optee_test/ta/create_fail_test/c3f6e2c0-3548-11e1-b86c0800200c9a66.elf
    /home/brand/devel/optee/toolchains/aarch32/bin/../lib/gcc/arm-linux-gnueabihf/4.9.2/libgcc.a(_dvmd_lnx.o): In function `__aeabi_ldiv0':
    (.text+0x6): undefined reference to `raise'
    /home/brand/devel/optee/toolchains/aarch32/bin/arm-linux-gnueabihf-ld: error: required section '.got' not found in the linker script
    /home/brand/devel/optee/toolchains/aarch32/bin/arm-linux-gnueabihf-ld: final link failed: Invalid operation

I think the TA links with the wrong division code: the one of libgcc (that uses `raise` - I guess that the code of A15 does not need the raise function) is used, instead of the one of libutils. But then, it looks like the `raise` symbol is defined in `libc`, which results in the link error.


I think we do not need it any longer because the GP API indicates that only function of the API should be used. For example, `memcpy` should not be used in TA, but `TEE_FillMem` should. I agree some user may issue not linking with `libgcc` when they directly import some "normal world" code that is compiling well into a TA. But then nothing prevents them to link with `libgcc` in the Makefile of the TA.